Source/hydro/Castro_ctu_nd.F90

```diff
@@ -2,10 +2,1384 @@
+    real(rt), intent(inout) ::    q3(q3_lo(1):q3_hi(1),q3_lo(2):q3_hi(2),q3_lo(3):q3_hi(3),NGDNV)
+#endif
+
+#ifdef RADIATION
+    real(rt), intent(inout) :: rflux1(rf1_lo(1):rf1_hi(1),rf1_lo(2):rf1_hi(2),rf1_lo(3):rf1_hi(3),0:ngroups-1)
+#if AMREX_SPACEDIM >= 2
+    real(rt), intent(inout) :: rflux2(rf2_lo(1):rf2_hi(1),rf2_lo(2):rf2_hi(2),rf2_lo(3):rf2_hi(3),0:ngroups-1)
+#endif
+#if AMREX_SPACEDIM == 3
+    real(rt), intent(inout) :: rflux3(rf3_lo(1):rf3_hi(1),rf3_lo(2):rf3_hi(2),rf3_lo(3):rf3_hi(3),0:ngroups-1)
+#endif
+#endif
+#if AMREX_SPACEDIM <= 2
+    real(rt), intent(in) :: dloga(dloga_lo(1):dloga_hi(1),dloga_lo(2):dloga_hi(2),dloga_lo(3):dloga_hi(3))
+    real(rt), intent(in) :: area1(area1_lo(1):area1_hi(1),area1_lo(2):area1_hi(2),area1_lo(3):area1_hi(3))
+    real(rt), intent(in) :: vol(vol_lo(1):vol_hi(1),vol_lo(2):vol_hi(2),vol_lo(3):vol_hi(3))
+#endif
+#if AMREX_SPACEDIM == 2
+    real(rt), intent(in) :: area2(area2_lo(1):area2_hi(1),area2_lo(2):area2_hi(2),area2_lo(3):area2_hi(3))
+#endif
+
+    real(rt) :: dxinv, dyinv, dzinv
+    real(rt) :: dtdx, dtdy, dtdz, hdt
+    real(rt) :: cdtdx, cdtdy, cdtdz
+    real(rt) :: hdtdx, hdtdy, hdtdz
+
+    integer :: i, j, k, iwave, idim, n
+
+    real(rt), pointer :: dqx(:,:,:,:), dqy(:,:,:,:), dqz(:,:,:,:)
+
+    real(rt), pointer :: Ip(:,:,:,:,:,:), Im(:,:,:,:,:,:)
+    real(rt), pointer :: Ip_src(:,:,:,:,:,:), Im_src(:,:,:,:,:,:)
+    real(rt), pointer :: Ip_gc(:,:,:,:,:,:), Im_gc(:,:,:,:,:,:)
+
+    real(rt)        , pointer :: shk(:,:,:)
+
+    real(rt)        , pointer :: sxm(:,:,:), sym(:,:,:), szm(:,:,:)
+    real(rt)        , pointer :: sxp(:,:,:), syp(:,:,:), szp(:,:,:)
+
+    ! Left and right state arrays (edge centered, cell centered)
+    double precision, dimension(:,:,:,:), pointer :: &
+         qxm, qym, qzm, qxp, qyp, qzp, ql, qr, &
+         qmxy, qpxy, qmxz, qpxz, qmyx, qpyx, &
+         qmyz, qpyz, qmzx, qpzx, qmzy, qpzy, &
+         qxl, qxr, qyl, qyr, qzl, qzr
+
+    double precision, dimension(:,:,:,:), pointer:: &
+         fx, fy, fz, fxy, fxz, fyx, fyz, fzx, fzy
+
+#ifdef RADIATION
+    double precision, dimension(:,:,:,:), pointer:: &
+         rfx, rfy, rfz, rfxy, rfxz, rfyx, rfyz, rfzx, rfzy
+#endif
+
+    double precision, dimension(:,:,:,:), pointer:: &
+         qgdnvx, qgdnvy, qgdnvz, &
+         qgdnvxy, qgdnvxz, &
+         qgdnvyx, qgdnvyz, &
+         qgdnvzx, qgdnvzy
+
+    ! these will be the temporary arrays we actually allocate space for
+    double precision, dimension(:,:,:,:), pointer :: ftmp1, ftmp2, rftmp1, rftmp2
+    double precision, dimension(:,:,:,:), pointer :: qgdnvtmp1, qgdnvtmp2
+
+    type (eos_t) :: eos_state
+
+    logical :: source_nonzero(QVAR)
+    logical :: reconstruct_state(NQ)
+
+    integer :: fglo(3), fghi(3), glo(3), ghi(3)
+
+    ! Local constants
+    dxinv = ONE/dx(1)
+    dtdx = dt*dxinv
+    hdtdx = HALF*dtdx
+    cdtdx = dtdx*THIRD
```

Comment:
Does this factor depend on AMREX_SPACEDIM or is it 1/3 independent of dimensionality?